astroplan/tests/test_constraints.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from ..constraints import (AltitudeConstraint, AirmassConstraint, AtNightConstraint,
+                           is_observable, is_always_observable, observability_table,
+                           time_grid_from_range, SunSeparationConstraint,
+                           MoonSeparationConstraint, MoonIlluminationConstraint,
+                           LocalTimeConstraint)
+from ..core import Observer, FixedTarget
+import numpy as np
+import astropy.units as u
+from astropy.time import Time
+from astropy.coordinates import SkyCoord, get_sun
+from ..moon import get_moon
+import datetime as dt
+
+vega = FixedTarget(coord=SkyCoord(ra=279.23473479*u.deg, dec=38.78368896*u.deg),
+                   name="Vega")
+rigel = FixedTarget(coord=SkyCoord(ra=78.63446707*u.deg, dec=8.20163837*u.deg),
+                    name="Rigel")
+polaris = FixedTarget(coord=SkyCoord(ra=37.95456067*u.deg,
+                                     dec=89.26410897*u.deg), name="Polaris")
+
+def test_at_night_basic():
+    subaru = Observer.at_site("Subaru")
+    time_ranges = [Time(['2001-02-03 04:05:06', '2001-02-04 04:05:06']), # 1 day
+                   Time(['2007-08-09 10:11:12', '2007-08-09 11:11:12'])] # 1 hr
+    targets = [vega, rigel, polaris]
+    for time_range in time_ranges:
+        # Calculate constraint using methods on astroplan.Observer:
+        observer_is_night = [subaru.is_night(time)
+                             for time in time_grid_from_range(time_range)]
+        observer_is_night_any = any(observer_is_night)
+        observer_is_night_all = all(observer_is_night)
+
+        assert all(is_observable(AtNightConstraint(), subaru, targets,
+                                 time_range=time_range) ==
+                   len(targets)*[observer_is_night_any])
+
+        assert all(is_always_observable(AtNightConstraint(), subaru, targets,
+                                        time_range=time_range) ==
+                   len(targets)*[observer_is_night_all])
+
+def test_observability_table():
+    subaru = Observer.at_site("Subaru")
+    time_ranges = [Time(['2001-02-03 04:05:06', '2001-02-04 04:05:06']), # 1 day
```

Comment:
My PyCharm also grays out the `time_ranges` variable, but it is being used [here](https://github.com/astropy/astroplan/pull/61/files#diff-9162b487e33fd6ae22fe36890f1827aeR53).